A32NX/html_ui/Pages/A32NX_Utils/SimBriefApi.js

```diff
@@ -1,17 +1,35 @@
 class SimBriefApi {
-    static getSimBriefOfp(username, userId) {
-        let apiUrl;
-        if (username) {
-            apiUrl = `${SimBriefApi.url}&username=${username}`;
-        } else if (userId) {
-            apiUrl = `${SimBriefApi.url}&userid=${userId}`;
+    static getSimBriefOfp(userId) {
+        if (userId) {
+            return fetch(`${SimBriefApi.url}&userid=${userId}`)
+                .then((response) => {
+                    if (!response.ok) {
+                        throw (response);
```

Comment:
I've noted this in other places before, but I think this is a very unusual pattern. Usually you would be throwing `new Error` objects. We shouldn't expect code calling `getSimBriefOfp` to handle the error and inspect status codes or something.